0-module-and-module-source.emu

```diff
@@ -0,0 +1,650 @@
+                    1. If _ie_.[[ImportName]] is ~namespace-object~, then
+                      1. NOTE: This is a re-export of an imported module namespace object.
+                      1. Append _ee_ to _localExportEntries_.
+                    1. Else,
+                      1. NOTE: This is a re-export of a single name.
+                      1. Append the ExportEntry Record { [[ModuleRequest]]: _ie_.[[ModuleRequest]], [[ImportName]]: _ie_.[[ImportName]], [[LocalName]]: *null*, [[ExportName]]: _ee_.[[ExportName]] } to _indirectExportEntries_.
+                1. Else if _ee_.[[ImportName]] is ~all-but-default~, then
+                  1. Assert: _ee_.[[ExportName]] is *null*.
+                  1. Append _ee_ to _starExportEntries_.
+                1. Else,
+                  1. Append _ee_ to _indirectExportEntries_.
+              1. Let _async_ be _body_ Contains `await`.
+              1. NOTE: How should _hostDefined_ be populated if this operation is triggered by the user and whether or not _hostDefined_ should be used for sources created in user land.
+              1. Let _hostDefined_ be *undefined*.
+              1. Return Module Source Record { [[HasTLA]]: _async_, [[ECMAScriptCode]]: _body_, [[RequestedModules]]: _requestedModules_, [[ImportEntries]]: _importEntries_, [[LocalExportEntries]]: _localExportEntries_, [[IndirectExportEntries]]: _indirectExportEntries_, [[StarExportEntries]]: _starExportEntries_, [[HostDefined]]: _hostDefined_ }.
+            </emu-alg>
+
+            <emu-note type="editor">
+                A Module Source Record is the static portion of a Module Record. A refactor of the spec can make this separation more formal instead of just having both of them sharing the same set of Internal Slots.
+             </emu-note>
+        </emu-clause>
+
+    </emu-clause>
+
+    <emu-clause id="sec-modulesource-constructor">
+        <h1>The ModuleSource Constructor</h1>
+        <p>The ModuleSource constructor:</p>
+        <ul>
+            <li>is the intrinsic object <dfn>%ModuleSource%</dfn>.</li>
+            <li>is the initial value of the *"ModuleSource"* property of the global object.</li>
+            <li>is not intended to be called as a function and will throw an exception when called in that manner.</li>
+            <li>creates and initializes a new ModuleSource object when called as a constructor.</li>
+        </ul>
+
+        <emu-clause id="sec-modulesource">
+            <h1>ModuleSource ( _sourceText_ )</h1>
+            <p>When the `ModuleSource` function is called with argument _sourceText_, the following steps are taken:</p>
+            <emu-alg>
+                1. If NewTarget is *undefined*, throw a *TypeError* exception.
+                1. Let _O_ be ? OrdinaryCreateFromConstructor(NewTarget, *"%ModuleSource.prototype%"*, « [[ModuleSource]] »).
+                1. Let _body_ be ParseText(_sourceText_, |Module|).
```

Comment:
This step should do a coherent to convert _sourceText_ to a String.

Reply:
fixed by https://github.com/tc39/proposal-compartments/pull/76